packer/core.go

```diff
@@ -218,15 +219,33 @@ func (c *Core) BuildNames(only, except []string) []string {
 func (c *Core) generateCoreBuildProvisioner(rawP *template.Provisioner, rawName string) (CoreBuildProvisioner, error) {
 	// Get the provisioner
 	cbp := CoreBuildProvisioner{}
+
+	if !c.components.PluginConfig.Provisioners.Has(rawP.Type) {
+		err := fmt.Errorf(
+			"The provisioner %s is unknown by Packer, and is likely part of a plugin that is not installed.\n"+
+				"You may find the needed plugin along with installation instructions documented on the Packer integrations page.\n\n"+
+				"https://developer.hashicorp.com/packer/integrations?filter=%s",
+			rawP.Type,
+			strings.Split(rawP.Type, "-")[0],
+		)
+
+		if sugg := didyoumean.NameSuggestion(rawP.Type, c.components.PluginConfig.Builders.List()); sugg != "" {
```

Comment:
Should this be c.components.PluginConfig.Provisioners.List()?

Reply:
Yes it should! Good catch, I'll open a PR to fix it ASAP

Reply:
https://github.com/hashicorp/packer/pull/12708